CMSgov/easi-app
# EASI-1977 Client Change Contract Vehicle field to Contract Number on Intake Form

# EASI-1977

This work follows the frontend set of changes

## Changes and Description

- Validation schema change for number field
- Client query has contract number assigned adjacent to legacy vehicle
- Display the contract number and/or vehicle according to certain conditions, when reviewing the intake



## Testing
- Jests have updated mocks with contract number assigned next to the legacy vehicle
- Cypress intake request test will target the new contract number field
- Run and qa locally
  - Db clean & seed
  - See some reviewable intakes
    - legacy contract vehicle
      http://localhost:3000/system/2ed89f9f-7fd9-4e92-89d2-cee170a44d0d/view
    - no contract number or vehicle
      http://localhost:3000/system/38e46d77-e474-4d15-a7c0-f6411221e2a4/view
    - contract number
      http://localhost:3000/system/af7a3924-3ff7-48ec-8a54-b8b4bc95610b/view
  - Create a new intake request and qa the new "Contract number field"

## Notes

- Submission at the point of the SubmitIntake mutate can take up to ~30s, locally.
  Was mentioned https://github.com/CMSgov/easi-app/pull/1719#pullrequestreview-1070076188
  Local cypress test will fail after 4s
- Month | Day | Year field has a bug, unrelated to this pr
  https://jiraent.cms.gov/browse/EASI-2311
